src/resources/invoices-resource.js

```diff
@@ -1,136 +1,126 @@
-            return apiHandler.delete(`invoices/${id}/timeline/${messageId}`);
-        },
-
-        createTimelineComment({id, data}) {
-            return apiHandler.create(`invoices/${id}/timeline`, '', data);
-        },
-
-        getAllTransactionAllocations({id, limit = null, offset = null}) {
-            const params = {
-                limit,
-                offset,
-            };
-
-            return apiHandler.getAll(`invoices/${id}/transaction-allocations`, params);
-        },
-        
-        applyTransaction({id, data}) {
-            return apiHandler.post(`invoices/${id}/transaction`, data);
-        },
-    };
-};
+  return {
+    getAll({
+      filter = null,
+      sort = null,
+      limit = null,
+      offset = null,
+      q = null,
+      expand = null,
+    } = {}) {
+      const params = {filter, sort, limit, offset, q, expand};
+      return apiHandler.getAll(`invoices`, params);
+    },
+    create({id = '', data, expand = null}) {
+      const params = {expand};
+      return apiHandler.create(`invoices/${id}`, id, data, params);
+    },
+    get({id, Accept, expand = null}) {
+      const params = {Accept, expand};
+      return apiHandler.get(`invoices/${id}`, params);
+    },
+    update({id, data, expand = null}) {
+      const params = {expand};
+      return apiHandler.put(`invoices/${id}`, data, params);
+    },
+    getAllInvoiceItems({
+      id = null,
+      limit = null,
+      offset = null,
+      expand = null,
+    } = {}) {
+      const params = {limit, offset, expand};
+      return apiHandler.getAll(`invoices/${id}/items`, params);
+    },
+    createInvoiceItem({id, data}) {
+      return apiHandler.post(`invoices/${id}/items`, data);
+    },
+    issue({id, data}) {
+      return apiHandler.post(`invoices/${id}/issue`, data);
+    },
+    abandon({id}) {
+      return apiHandler.post(`invoices/${id}/abandon`);
+    },
+    void({id}) {
+      return apiHandler.post(`invoices/${id}/void`);
+    },
+    recalculate({id}) {
+      return apiHandler.post(`invoices/${id}/recalculate`);
+    },
+    reissue({id, data}) {
+      return apiHandler.post(`invoices/${id}/reissue`, data);
+    },
+    getAllTransactionAllocations({
+      id = null,
+      limit = null,
+      offset = null,
+      expand = null,
+    } = {}) {
+      const params = {limit, offset, expand};
+      return apiHandler.getAll(
+        `invoices/${id}/transaction-allocations`,
+        params
+      );
+    },
+    applyTransaction({id, data}) {
+      return apiHandler.post(`invoices/${id}/transaction`, data);
+    },
+    getAllTimelineMessages({
+      id = null,
+      limit = null,
+      offset = null,
+      filter = null,
+      sort = null,
+      q = null,
+      expand = null,
+    } = {}) {
+      const params = {limit, offset, filter, sort, q, expand};
+      return apiHandler.getAll(`invoices/${id}/timeline`, params);
+    },
+    createTimelineComment({id, data}) {
```

Comment:
On the "old" one there is an empty string `return apiHandler.create(`invoices/${id}/timeline`, '', data)` not sure what that means, maybe an error on the handmade one?

Reply:
Yeah, that was working but it was odd. 
Explanation: 
`apiHandler.create` calls `put` or `post` depending if you pass an `id` or not
We were always passing an empty `id` with means that can replace the old `apiHandler.create` with a `apiHandler.post` that does not need `id` parameter.

In this case: 🤖 won 🧑‍💻